library/src/main/java/com/getstream/sdk/chat/utils/LlcMigrationUtils.java

```diff
@@ -213,15 +213,7 @@ public static ChannelUserRead getRead(Channel channel, String userId) {
     public static boolean lastMessagesAreTheSame(Channel a, Channel b) {
         Message oldLastMessage = computeLastMessage(a);
         Message newLastMessage = computeLastMessage(b);
-        if (oldLastMessage != null &&
-                newLastMessage != null &&
-                newLastMessage.getUpdatedAt() != null &&
-                oldLastMessage.getUpdatedAt() != null &&
-                oldLastMessage.getUpdatedAt().getTime() < newLastMessage.getUpdatedAt().getTime()) {
-            return false;
-        } else {
-            return true;
-        }
+        return oldLastMessage == newLastMessage;
```

Comment:
simplify this check now that the message is a data class with correct comparison methods

Reply:
Do we need that method anymore if it's a straight equality check? Can we also just do a straight equality check in the diff callback's `areContentsTheSame` method?

Reply:
no, then again this entire file needs to be cleaned up/converted to kotlin. don't want to do it now.

Reply:
In Java, using `==` is a comparison of the references, and won't use the generated `equals` method of the data class. If we want to use `equals`, we also have to perform null checks on the messages first, because (at least according to the `computeLastMessage` method's declaration) either might be `null`, and calling `equals` on that would crash.

Reply:
how about we close this PR and move the whole thing to kotlin?

Reply:
well, the current version is broken, hmm.